tslib.es6.js

```diff
@@ -348,6 +348,15 @@ export function __disposeResources(env) {
     return next();
 }
 
+export function __rewriteRelativeImportExtension(path, preserveJsx) {
+    if (typeof path === "string" && /^\.\.?\//.test(path)) {
+        return path.replace(/\.(tsx)$|(\.d)?(\.[^./]+?)?\.([cm])?ts$/i, function (m, tsx, d, ext, cm) {
+            return tsx ? preserveJsx ? ".jsx" : ".js" : d && (ext && !cm || !ext) ? m : ((d || "") + (ext || "") + "." + (cm || "").toLowerCase() + "js");

```

Comment:
```suggestion
            return tsx ? preserveJsx ? ".jsx" : ".js" : d && (!ext || !cm) ? m : ((d || "") + (ext || "") + "." + (cm || "").toLowerCase() + "js");
```

To reduce the amount of `|| ""` that you're doing, you could move the quantifiers inside of the capture groups so that they capture empty strings instead of `undefined`:

```
/\.(tsx)$|((?:\.d)?)((?:\.[^./]+?)?)\.([cm]?)ts$/i
```

Also, to clarify the `d && (ext && !cm || !ext)` case Is this saying that `d.ts` (no `ext` and no `cm`) remains `.d.ts`, `.d.cts` (no `ext` but has `cm`) remains `.d.cts`, but `.d.css.cts` becomes `.d.css.cjs`?

Reply:
Yes, that’s right. `.d.css.cts` is not a recognized declaration file extension (even though it looks suspiciously like one), so we would have to treat it as any other `*.cts` file name. A real fishy edge case, but correct for today’s compiler, anyway.